src/Array.ts

```diff
@@ -14,8 +14,9 @@ import { Plus1 } from './Plus'
 import { Setoid, getArraySetoid } from './Setoid'
 import { Traversable1 } from './Traversable'
 import { Unfoldable1 } from './Unfoldable'
-import { Endomorphism, Predicate, Refinement, concat, identity, tuple } from './function'
-import { Compactable1, Separated } from './Compactable'
+import { Endomorphism, Predicate, Refinement, concat, tuple } from './function'
+import { Compactable1, getCompactFromFilterMap, getSeparateFromPartitionMap } from './Compactable'
```

Comment:
I wouldn't use default implementations for the core data structures

Reply:
Ok, I'll update all ADTs with efficient versions

Reply:
> Ok, I'll update all ADTs with efficient versions

May I help you? I'm playing with some implementations. Looks like `compact` and `separate` can be derived efficiently from (efficient) `filterMap` and `partitionMap`, no matter what data structure

```
compact = fa => filterMap(fa, identity)
separate = fa => partitionMap(fa, identity)
```